src/charm.py

```diff
@@ -112,15 +131,41 @@ def _configure(self, _):
             reloaded = self._prometheus_server.reload_configuration()
             if not reloaded:
                 self.unit.status = BlockedStatus("Failed to load Prometheus config")
+                return
             else:
                 self.unit.status = ActiveStatus()
                 logger.info("Prometheus configuration reloaded")
         else:
             container.add_layer(self._name, new_layer, combine=True)
             container.restart(self._name)
             logger.info("Prometheus (re)started")
+
+        remote_write_plus_ingress_status_message = INGRESS_MULTIPLE_UNITS_STATUS_MESSAGE
+
+        if not self._validate_ingress_and_remote_write():
+            self.unit.status = BlockedStatus(remote_write_plus_ingress_status_message)
+            logger.error(
+                "Using the ingress relation and receiving remote_write relations with more than "
+                "one Prometheus unit; remote_write data is likely to be sent to only one unit."
+            )
+        elif (
+            isinstance(self.unit.status, BlockedStatus)
+            and self.unit.status.message == remote_write_plus_ingress_status_message
+        ):
+            self.unit.status = ActiveStatus()
```

Comment:
Afaiu the pattern in _configure is to return immediately after status is set to something other than active, and only at the bottom of the function set the status to active. Not sure if this status check with string comparison is needed.